asapdiscovery-ml/asapdiscovery/ml/inference/inference.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from pathlib import Path
+from typing import Dict, Optional
+
+import torch
+
+# static import of models from base yaml here
+from asapdiscovery.ml.pretrained_models import all_models
+from asapdiscovery.ml.utils import build_model, load_weights
+from asapdiscovery.ml.weights import ModelSpec, fetch_file, fetch_model_from_spec
+
+
+class InferenceBase:
+    """
+    Inference base class for PyTorch models in asapdiscovery.
+
+    Parameters
+    ----------
+    model_name : str
+        Name of model to use.
+    model_type : str
+        Type of model to use.
+    from_spec : bool, default=True
+        Whether to fetch weights from asapdiscovery.ml spec file or from local file.
+    build_model_kwargs : Optional[Dict], default=None
+        Keyword arguments to pass to build_model function.
+    device : str, default='cpu'
+        Device to use for inference.
+
+
+    Methods
+    -------
+    predict
+        Predict on a batch of data.
+
```

Comment:
add `build_model` method
```suggestion
    build_model
        Build model from arguments

```